dashboard/src/components/Header/ClusterSelector.test.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react";
+import * as ReactRedux from "react-redux";
+
+import { getStore, mountWrapper } from "shared/specs/mountWrapper";
+import ClusterSelector, { IClusterSelectorProps } from "./ClusterSelector";
+
+const defaultProps = {
+  onChange: jest.fn(),
+  clusters: {
+    currentCluster: "default",
+    clusters: {
+      default: {
+        currentNamespace: "default",
+        namespaces: ["default", "other"],
+      },
+      other: {
+        currentNamespace: "default",
+        namespaces: ["default", "other"],
+      },
+    },
+  },
+} as IClusterSelectorProps;
+
+it("renders the cluster selector with the correct options", () => {
+  const store = getStore({});
+
+  const wrapper = mountWrapper(store, <ClusterSelector {...defaultProps} />);
+  const options = wrapper.find("Select").prop("options");
+
+  expect(options).toEqual([
+    { label: "default", value: "default" },
+    { label: "other", value: "other" },
+  ]);
+});
+
+it("dispatches fetchNamespaces and calls onChange prop when changed", () => {
+  const store = getStore({});
+  const onChange = jest.fn();
+  const props = {
+    ...defaultProps,
+    onChange,
+  };
+  const mockDispatch = jest.fn();
+  const spyOnUseDispatch = jest.spyOn(ReactRedux, "useDispatch").mockReturnValue(mockDispatch);
```

Comment:
:+1: `jest.spyOn` was what I was looking for. BTW, you may want to move this to a `beforeEach` method to avoid copying it for every test.

Reply:
Yep, if there was ore than one test using it.